### ISSUE TITLE
Download 1.15.2-8.0.0-SNAPSHOT when api version targeting `8.1.0-SNPSHOT`

### ISSUE THREAD

**Issue by SettingDust** [SpongePowered/SpongeGradle#37]
SG: 2.0.1
Error: `Failed to find Premain-Class manifest attribute in C:\Users\setti\.gradle\caches\modules-2\files-2.1\org.spongepowered\spongevanilla\1.15.2-8.0.0-SNAPSHOT\9c0fa653fdab9b3e9af951086effcbbd485acbf2\spongevanilla-1.15.2-8.0.0-SNAPSHOT-universal.jar`

**Comment by zml2008** [comment]
This seems to be an issue with Sponge, where Sponge is publishing 8.1.0-SNAPSHOT versions under 8.0.0.

The logic in `generateApiReleasedVersion` seems to do this intentionally on the Sponge side. Should we match this in SG, or modify sponge to just strip the -SNAPSHOT off API versions?

```java
    private static String generateApiReleasedVersion(final String apiVersion) {
        final String[] apiSplit = apiVersion.replace("-SNAPSHOT", "").split("\\.");
        final boolean isSnapshot = apiVersion.contains("-SNAPSHOT");

        // This is to determine if the split api version has at the least a minimum version.
        final String apiMajor = apiSplit[0];
        final String minorVersion;
        if (apiSplit.length > 1) {
            minorVersion = apiSplit[1];
        } else {
            minorVersion = "0";
        }
        final int latestReleasedVersion = Math.max(Integer.parseInt(minorVersion) - 1, 0);
        // And then here, we determine if the api version still has a patch version, to just ignore it.
        final String latestReleasedApiMinor = isSnapshot ? String.valueOf(latestReleasedVersion) : minorVersion;
        return apiMajor + "." + latestReleasedApiMinor;
    }
```

thoughts @gabizou?

**Comment by gabizou** [comment]
> Should we match this in SG

In short, yes.

**Comment by zml2008** [comment]
Fixed in e0176ccdab98e1c7dc1e5bdd34148ecd5cc28765
